tests/test_spider.py

```diff
@@ -418,6 +418,9 @@ def test_get_sitemap_body(self):
         r = Response(url="http://www.example.com/favicon.ico", body=self.BODY)
         self.assertSitemapBody(r, None)
 
+        r = Response(url="http://www.example.com", body=b"")
+        self.assertSitemapBody(r, None)
```

Comment:
Hi @Endi1 thank you for adding a test!!

could you add a test that verifies that this warning is indeed logged? It should be as easier as doing:
```
with warnings.catch_warnings(record=True) as w:
    r = Response(url="http://www.example.com", body=b"")
    self.assertSitemapBody(r, None)

    self.assertEqual(len(w), 1)
    self.assertIn("Ignoring invalid sitemap", w[0].message)
```

Thanks in advance!!

Reply:
Ah, and you can also use `self.assertWarnsRegex` and `self.assertWarns` (if you prefer it).

Reply:
Doesn't `warnings.catch_warning` catch only warnings issued with `warnings.warn()`? I just tested this to be certain and sure enough, the test fails when I add this assertion.

Reply:
Hi @Endi1 

There is something that I can't understand by reading your code / tests. The test you added is for the `_get_sitemap_body` method instead of the `_parse_sitemap` you changed (and it's not called from the other method). If the test you added works, that means that even when having an empty response the  `_get_sitemap_body` method would return `None` and your changes wouldn't be necessary (because there is a `if body is None:` ...).

What do you think? It could be a good idea to do TDD here. It could be better to be able to reproduce the error and get a failing test before changing the code and then fix it. 

@Gallaecio could you give a hand here?

Reply:
Yes, it’s not warnings that you need to capture, but log messages. There is a `LogCapture` class in the tests that should do the trick, and works similarly to @noviluni ’s warning capture example.

However, @Endi1 I don’t think your test reaches your new code. Your change is in `_parse_sitemap`, while the test class you are using calls `_get_sitemap_body` directly-